src/cc/bcc_syms.cc

```diff
@@ -177,13 +207,21 @@ bool ProcSyms::Module::is_perf_map() const {
 }
 
 void ProcSyms::Module::load_sym_table() {
+  struct ns_cookie nsc = {-1, -1};
+
   if (syms_.size())
     return;
 
-  if (is_perf_map())
+  if (is_perf_map()) {
+    if (in_ns_)
+      bcc_procutils_enter_mountns(pid_, &nsc);
     bcc_perf_map_foreach_sym(name_.c_str(), _add_symbol, this);
-  else
+  } else {
+    bcc_procutils_enter_mountns(pid_, &nsc);
     bcc_elf_foreach_sym(name_.c_str(), _add_symbol, this);
+  }
+
+  bcc_procutils_exit_mountns(&nsc);
```

Comment:
Ah, sorry.  Two different things going on here.  In the case of a perf-map, we want to try to look in the mount namespace if we've already determined that a perf.map is there.  If not, we'll double-check on the host container.  For standard ELF symbols, this code always attempts to look in the target namespace.